src/net_processing.cpp

```diff
@@ -4549,7 +4765,39 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                         if (pto->m_tx_relay->pfilter && !pto->m_tx_relay->pfilter->IsRelevantAndUpdate(*txinfo.tx)) continue;
                         // Send
                         State(pto->GetId())->m_recently_announced_invs.insert(hash);
-                        vInv.push_back(inv);
+
+                        bool added_to_recon_set = false;
+                        // Check if peer supports reconciliations.
+                        if (supports_recon) {
+                            // Check if reconciliation set is not at capacity for two reasons:
+                            // - limit sizes of reconciliation sets and short id mappings
+                            // - limit CPU use for sketch computations
+                            //
+                            // Since we reconcile frequently, reaching capacity either means:
+                            // (1) a peer for some reason does not request reconciliations from us for a long while, or
+                            // (2) really a lot of valid fee-paying transactions were dumped on us at once.
+                            // We don't care about a laggy peer (1) because we probably can't help them even if we flood transactions.
+                            // However, exploiting (2) should not prevent us from relaying certain transactions.
+                            //
+                            // Transactions which don't make it to the set due to the limit are announced via flooding.
+                            const size_t recon_set_size = *m_reconciliation.GetPeerSetSize(pto->GetId());
+                            if (txs_to_reconcile.size() + recon_set_size < MAX_PEER_TX_ANNOUNCEMENTS) {
```

Comment:
I don't think `MAX_PEER_TX_ANNOUNCEMENTS` is covered by `p2p_erlay.py` ? This diff doesn't yell an error.

```
diff --git a/src/net_processing.cpp b/src/net_processing.cpp
index 689b899f1..0703fcea7 100644
--- a/src/net_processing.cpp
+++ b/src/net_processing.cpp
@@ -81,7 +81,7 @@ static constexpr int32_t MAX_PEER_TX_REQUEST_IN_FLIGHT = 100;
  *  the actual transaction (from any peer) in response to requests for them.
  *  Also limits a maximum number of elements to store in the reconciliation set.
  */
-static constexpr int32_t MAX_PEER_TX_ANNOUNCEMENTS = 5000;
+static constexpr int32_t MAX_PEER_TX_ANNOUNCEMENTS = 4999;
 /** How long to delay requesting transactions via txids, if we have wtxid-relaying peers */
 static constexpr auto TXID_RELAY_DELAY = std::chrono::seconds{2};
 /** How long to delay requesting transactions from non-preferred peers */
```

Reply:
I tried to make this test, but signing 5000 transactions takes minutes (that's expected, right?), and that's the only way to test this.

Instead, I'm adding tests for `MAX_SKETCH_CAPACITY = 2 << 12`